docs/se/guides/migration_3x.adoc

```diff
@@ -0,0 +1,266 @@
+
+.Enabling Http/2 support in Helidon 3 by adding dependency
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.webserver</groupId>
+    <artifactId>helidon-webserver-http2</artifactId>
+</dependency>
+----
+
+With above dependency Helidon 3 supports Http/2 upgrade from Http/1,
+cleartext Http/2 without prior knowledge and Http/2 with ALPN over TLS.
+
+In Helidon 2, max content length was configurable with `server.experimental.http2-max-content-length`, in Helidon 3 can be configured with `server.max-upgrade-content-length` globally or per socket with the same `max-upgrade-content-length` key.
+
+.Max upgrade content length in Helidon 3
+[source,yaml]
+----
+server:
+  port: 8080
+  host: 0.0.0.0
+  max-upgrade-content-length: 16384
+----
+
+For further information check xref:../webserver.adoc[WebServer Documentation]
+
+== WebSocket
+
+Helidon SE support is now based on the `WebSocketRouting` class which enables Helidon application to
+configure routing for both annotated and programmatic WebSocket endpoints.
+`TyrusSupport` is now deprecated. Websocket support in now placed in different artifact.
+
+.Helidon 2 WebSocket support dependency
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.webserver</groupId>
+    <artifactId>helidon-webserver-tyrus</artifactId>
+</dependency>
+----
+
+.Helidon 3 WebSocket support dependency
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.webserver</groupId>
+    <artifactId>helidon-webserver-websocket</artifactId>
+</dependency>
+----
+
+In Helidon 2, WebSocket routing is defined by registering `TyrusSupport` as additional service:
+
+.Helidon 2 WebSocket route registering
+[source, java]
+----
+WebServer.builder(Routing.builder()
+             .register("/rest", new SomeRestService()) //<1>
+             .register("/websocket",TyrusSupport.builder() //<2>
+                 .register(ServerEndpointConfig.Builder
+                           .create(MessageBoardEndpoint.class, "/")
+                           .encoders(encoders)
+                           .build())
+                 .build()
+             ))
+             .port(8080)
+   .build();
+----
+<1> Traditional REST routing service registration
+<2> WebSocket setup with Tyrus service
+
+In Helidon 3, WebSocket routing is defined by adding another routing:
+
+.Helidon 3 WebSocket route registering
+[source, java]
+----
+WebServer.builder()
+         .routing(r -> r
+              .register("/rest", new SomeRestService()) //<1>
+         )
+         .addRouting(WebSocketRouting.builder() //<2>
+                        .endpoint("/websocket", ServerEndpointConfig.Builder
+                                .create(MessageBoardEndpoint.class, "/board")
+                                .encoders(encoders)
+                                .build())
+                        .build())
+         .port(8080)
+----
+<1> Traditional REST routing service registration
+<2> WebSocket routing setup
+
+
+== Deprecations
+
```

Comment:
Do we really give links to our PRs? Perhaps this is a placeholder for the doc once its live on Helidon.io?

Reply:
this is what we agreed